providers/base/fetch.go

```diff
@@ -101,15 +101,24 @@ func (p *BaseProvider[K, V]) attemptDataUpdate(ctx context.Context, responseCh c
 
 // startWebSocket starts a connection to the web socket and handles the incoming messages.
 func (p *BaseProvider[K, V]) startWebSocket(ctx context.Context, responseCh chan<- providertypes.GetResponse[K, V]) error {
-	p.logger.Debug("starting web socket query handler")
+	// Start the web socket query handler. If the connection fails to start, then the query handler
+	// will be restarted after a timeout.
+	for {
+		select {
+		case <-ctx.Done():
+			p.logger.Info("provider stopped via context")
+			return ctx.Err()
+		default:
+			p.logger.Debug("starting web socket query handler")
+			if err := p.ws.Start(ctx, p.ids, responseCh); err != nil {
+				p.logger.Error("web socket query handler returned error", zap.Error(err))
+			}
 
-	if err := p.ws.Start(ctx, p.ids, responseCh); err != nil {
-		p.logger.Error("web socket query handler returned error", zap.Error(err))
-		return err
+			// If the web socket query handler returns, then the connection was closed. Wait for
+			// a bit before trying to reconnect.
+			time.Sleep(p.cfg.WebSocket.ReconnectionTimeout)
```

Comment:
What happens if the remote client doesn't close the connection. I think the read will block indefinitely unless we [SetReadDeadline](https://pkg.go.dev/github.com/gorilla/websocket#Conn.SetReadDeadline).

Reply:
gonna add this as another thing we need to configure on the web socket providers.

Reply:
https://linear.app/skip/issue/BLO-758/figure-out-what-the-best-default-dial-configurations-are-for-web